renv.Rmd

```diff
@@ -3,95 +3,117 @@
+This approach ensures that the set of dependencies is collaboratively shared and maintained. Moreover, it can be used to align the development environment to the production stage. Managing and controlling dependencies all the way up from development to production improves reproducibility and empowers automated deployments, a key requirement in enterprise.
+
+Finally, as the control is at project level, it is possible to have different virtual environments (with different packages version) for different projects, while keeping the maintenance of the versioned package library efficient through caching.
 
 ## Install `renv`
 
 `renv` can be installed from CRAN as:
 
-```{r, echo =  TRUE , eval = FALSE}
+```{r, echo = TRUE, eval = FALSE}
 install.packages("renv")
 ```
 
-## Initialize an `renv` project
+## Set-up an `renv` project
 
-To initialize a package with `renv`, run:
+In order resolve and track dependencies of a project, `renv` needs to discover what the required packages are. Although `renv` supports discovering dependencies by scanning all files in the project (_implicit mode_), we strongly advise to define the required dependencies in a `DESCRIPTION` file, and rely on `renv`'s _explicit mode_ (see ['Snapshot types'](https://rstudio.github.io/renv/reference/snapshot.html#snapshot-types) in the documentation for details).
 
-```{r , echo = TRUE, eval = FALSE}
-renv::init(
-  # use the DESCRIPTION file to capture dependencies
-  settings = list(snapshot.type = "explicit"),
-  # do not install dependencies (done in a custom way)
-  bare = TRUE
-)
-```
+To initialize `renv` for a project where dependencies are explicitly stated in the `DESCRIPTION` file, run:
 
-This ensures that the dependencies of the package, as listed in the `DESCRIPTION` file are used for initializing the virtual environment. 
+```{r, echo = TRUE, eval = FALSE}
+renv::init(settings = list(snapshot.type = "implicit"))
+```
 
-Note: `renv` tracks and installs only the packages explicitly mentioned in the `DESCRIPTION` file. This means that only the dependencies used by the package are installed. Packages needed for development, such as `devtools`, should be be installed for the specific project via `install.packages()` or `renv::install()`.
+This ensures that the dependencies of the projects, listed in the `DESCRIPTION` file, are considered for initializing the virtual environment.
 
 The `renv` initialization causes:
 
-- the creation of a new `renv.lock` file where the R version and the list of used packages with their version are tracked;
+- the creation of the lock-file `renv.lock`, which tracks the version of the resolved transitive package dependencies, along with the R version and package repositories information;
+- the set-up of `renv` infrastructure for the project including a specific library;
```

Comment:
Correct, I think we can phrase it as follows, since the part about "the generation of the list of packages currently being used" is covered by the generation of `renv.lock`

```suggestion
- the set-up of `renv` infrastructure for the project including a specific library, where packages are installed at the versions tracked in `renv.lock`;
```